app/registries/foreman/plugin.rb

```diff
@@ -38,21 +38,43 @@ class PluginRequirementError < Foreman::Exception; end
   #   end
   #
   class Plugin
+    DEFAULT_REGISTRIES = {
+      fact_importer: -> { Plugin::FactImporterRegistry.new },
+      report_scanner: -> { Plugin::ReportScannerRegistry.new },
+      report_origin: -> { Plugin::ReportOriginRegistry.new },
+      medium_providers: -> { Plugin::MediumProvidersRegistry.new },
+      graphql_types: -> { Plugin::GraphqlTypesRegistry.new },
+    }
+
     @registered_plugins = {}
     @tests_to_skip = {}
-    @fact_importer_registry = Plugin::FactImporterRegistry.new
-    @report_scanner_registry = Plugin::ReportScannerRegistry.new
-    @report_origin_registry = Plugin::ReportOriginRegistry.new
-    @medium_providers = Plugin::MediumProvidersRegistry.new
-    @graphql_types_registry = Plugin::GraphqlTypesRegistry.new
+    @registries = {}
 
     class << self
       attr_reader   :registered_plugins
-      attr_accessor :tests_to_skip, :report_scanner_registry,
-        :report_origin_registry, :medium_providers,
-        :graphql_types_registry, :fact_importer_registry
+      attr_reader   :registries
+      attr_accessor :tests_to_skip
       private :new
 
+      # Defines a global registry to be used by other plugins as Foreman::Plugin.<global_registry_name>.
+      #
+      # Require a name of the registry and the registry instance.
+      # Registry name has to have _registry suffix
+      # ==== Examples
+      #
+      #   global_registry(:my_special_registry, MyPluginNamespace::MySpecialRegistry.new)
+      def global_registry(name, registry)
+        raise "Registry name mustn't have '_registry' suffix it is added automatically, you've provided #{name}" if name.to_s =~ /registry$/
```

Comment:
I'm a big fan of using `.start_with?` and `.end_with?` instead of regexes
```suggestion
        raise "Registry name mustn't have '_registry' suffix it is added automatically, you've provided #{name}" if name.to_s.end_with?('_registry')
```